pkg/verifier/gcp/entry_point.go

```diff
@@ -23,52 +26,93 @@ const (
 // - find unreachable endpoints & parse output, then terminate instance
 // - return `g.output` which stores the execution results
 func (g *GcpVerifier) ValidateEgress(vei verifier.ValidateEgressInput) *output.Output {
-	g.Logger.Debug(vei.Ctx, "Using configured timeout of %s for each egress request", vei.Timeout.String())
-	//default gcp machine e2
+	// Validate cloud platform type and default to PlatformGCP if not specified
+	if vei.PlatformType == "" {
+		vei.PlatformType = helpers.PlatformGCP
+	}
+	if !vei.CPUArchitecture.IsValid() {
+		vei.CPUArchitecture = cpu.ArchX86
+	}
+	// Default to curl.Probe if no Probe specified
+	if vei.Probe == nil {
+		vei.Probe = curl.Probe{}
+		g.Logger.Debug(vei.Ctx, "defaulted to curl probe")
+	}
+
+	// Set timeout to default if not specified
+	if vei.Timeout <= 0 {
+		vei.Timeout = DEFAULT_TIMEOUT
+	}
+	g.Logger.Debug(vei.Ctx, "configured a %s timeout for each egress request", vei.Timeout)
+
+	// Set instance type to default if not specified and validate it
 	if vei.InstanceType == "" {
 		vei.InstanceType = DEFAULT_INSTANCE_TYPE
 	}
```

Comment:
```suggestion
	// Set instance type to default if not specified and validate it
	if vei.InstanceType == "" {
		vei.InstanceType, err = vei.CPUArchitecture.DefaultInstanceType(helpers.PlatformGCP)
		if err != nil {
			return g.Output.AddError(err)
		}
		g.Logger.Debug(vei.Ctx, fmt.Sprintf("defaulted to instance type %s", vei.InstanceType))
	}
```
Use `vei.CPUArchitecture.DefaultInstanceType()` instead of a `DEFAULT_INSTANCE_TYPE` constant in order to enable ARM-friendliness. We also probably need to do a little work on lining up the "user specified a bad instance type; what do?" logic with how it's done on the AWS side, but this works okay for now.